runtime/src/tiered_storage.rs

```diff
@@ -252,4 +270,41 @@ mod tests {
             )),
         );
     }
+
+    #[test]
+    fn test_remove_on_drop() {
+        // Generate a new temp path that is guaranteed to NOT already have a file.
+        let temp_dir = tempdir().unwrap();
+        let tiered_storage_path = temp_dir.path().join("test_remove_on_drop");
+        {
+            let tiered_storage =
+                TieredStorage::new_writable(&tiered_storage_path, HOT_FORMAT.clone());
+            write_zero_accounts(&tiered_storage, Err(TieredStorageError::Unsupported()));
+        }
+        // expect error as the file has been removed on drop
+        assert!(fs::metadata(&tiered_storage_path).is_err());
```

Comment:
Done!

Somehow it does not allow me to do a direct comparison (possibly because Result doesn't implement PartialEq).  Have changed it to `assert!(!tiered_storage_path.try_exists().unwrap())` instead.

Reply:
Right.  I think we are good here as `unwrap()` covers `Ok` part, and pure `assert!` covers the boolean.

We can't do `assert_eq!` here with `Ok(true)` or `Ok(false)` because operation `==` cannot be applied to Result:

```
error[E0369]: binary operation `==` cannot be applied to type `std::result::Result<bool, std::io::Error>`
   --> runtime/src/tiered_storage.rs:279:9
    |
279 |         assert_eq!(tiered_storage_path.try_exists(), Ok(false));
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |         |
    |         std::result::Result<bool, std::io::Error>
    |         std::result::Result<bool, std::io::Error>
```